pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,22 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// TaintNotToleratedWithFilter checks if the given tolerations tolerates
+// all the filtered taints, and returns the first taint without a toleration
+func TaintNotToleratedWithFilter(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) *v1.Taint {
```

Comment:
How about FindMatchingUntoleratedTaint?

Reply:
Honestly this is a hard to understand function, I'd expect this to be two functions that are easy to compose. One to filter, one to check toleration. Although the second is a trivial loop, so I'm not sure if it really needs a dedicated function.

Reply:
I've changed this